src/nbl/video/IOpenGLSyncPrimitiveBase.h

```diff
@@ -25,7 +25,7 @@ class IOpenGLSyncPrimitiveBase
         if (!m_sync)
             m_toBeSignaled = true;
     }
-    // Answer to "what if the sync is waited upon before it was signaled (before actual GLsync exist) ?"
+    // Answer to "what if the sync is waited upon before actual GLsync exist ?"
```

Comment:
the design of this class is broken

Reply:
?

Reply:
it doesnt do the refcounting to a single `GLSync` object.

Reply:
it holds COpenGLSync which does the refcounting

Reply:
but you dont pass an `smart_refctd_ptr<COpenGLSync>` to signal a fence/semaphore/event (and to unsignal an event as well)

COpenGLSync makes its own GLSync inside

which makes it pointless to have a refcounted sync in the first place.

The aim to have only one `GLsync` be created by the queue as part of a submit (one for all semaphores and the fence).

Reply:
> but you dont pass an smart_refctd_ptr<COpenGLSync> to signal a fence/semaphore/event

how not? thats the point of having COpenGLSync